wagtail/wagtailadmin/decorators.py

```diff
@@ -3,26 +3,33 @@
 from django.contrib.auth.views import redirect_to_login as auth_redirect_to_login
 from django.core.urlresolvers import reverse
 from django.utils.translation import ugettext as _
+from django.http import HttpResponseForbidden
 
 from wagtail.utils.compat import user_is_anonymous
 from wagtail.wagtailadmin import messages
 
 
-def redirect_to_login(request):
-    return auth_redirect_to_login(
-        request.get_full_path(), login_url=reverse('wagtailadmin_login'))
+def reject_request(request):
+    if request.is_ajax():
+        return HttpResponseForbidden()
+    else:
+        return auth_redirect_to_login(
+            request.get_full_path(), login_url=reverse('wagtailadmin_login'))
 
 
 def require_admin_access(view_func):
     def decorated_view(request, *args, **kwargs):
         user = request.user
 
         if user_is_anonymous(user):
-            return redirect_to_login(request)
+            return reject_request(request)
 
         if user.has_perms(['wagtailadmin.access_admin']):
             return view_func(request, *args, **kwargs)
 
-        messages.error(request, _('You do not have permission to access the admin'))
-        return redirect_to_login(request)
+        if not request.is_ajax():
```

Comment:
This bit of code adds a message to display to the user when they get redirected to the dashboard but AJAX requests will no longer be redirected so I removed the message to prevent it from popping up in confusing places.